cluster/ubuntu/download-release.sh

```diff
@@ -52,8 +52,21 @@ grep -q "^${ETCD_VERSION}\$" binaries/.etcd 2>/dev/null || {
   echo ${ETCD_VERSION} > binaries/.etcd
 }
 
+function get_latest_version_number {
+  local -r latest_url="https://storage.googleapis.com/kubernetes-release/release/stable.txt"
+  if [[ $(which wget) ]]; then
+    wget -qO- ${latest_url}
+  elif [[ $(which curl) ]]; then
+    curl -Ss ${latest_url}
+  else
+    echo "Couldn't find curl or wget.  Bailing out."
+    exit 4
+  fi
+}
+
+KUBE_VERSION=`echo $(get_latest_version_number) | sed 's/v//'`
```

Comment:
Done.

Since you suggested these changes, I'd also try to push them to the shell
script that resides below, on which my change was originally based upon.

https://get.k8s.io/


On Mon, May 9, 2016 at 1:55 PM, Mike Danese <notifications@github.com>
wrote:

> In cluster/ubuntu/download-release.sh
> <https://github.com/kubernetes/kubernetes/pull/25366#discussion_r62571121>
> :
>
> > @@ -52,8 +52,21 @@ grep -q "^${ETCD_VERSION}\$" binaries/.etcd 2>/dev/null || {
> >    echo ${ETCD_VERSION} > binaries/.etcd
> >  }
> >
> > +function get_latest_version_number {
> > +  local -r latest_url="https://storage.googleapis.com/kubernetes-release/release/stable.txt"
> > +  if [[ $(which wget) ]]; then
> > +    wget -qO- ${latest_url}
> > +  elif [[ $(which curl) ]]; then
> > +    curl -Ss ${latest_url}
> > +  else
> > +    echo "Couldn't find curl or wget.  Bailing out."
> > +    exit 4
> > +  fi
> > +}
> > +
> > +KUBE_VERSION=`echo $(get_latest_version_number) | sed 's/v//'`
>
> also if you just want to remove leading v, then you should use a more
> explicit match expression, e.g. 's/^v//'
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25366/files/fc88b7b6c469ea32415366ec12ec85a200a0d1ac#r62571121>
>



-- 
This confidential email is read by the NSA (at the least).